### ISSUE TITLE
UI stops updating after websocket reconnect

### ISSUE THREAD

**Issue by amigan** [home-assistant/frontend#11038]
### Checklist

- [x] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Beginning with the update to 2021.12.6/home-assistant-frontend-20211227.0, after the websocket times out and reconnects, the UI stops updating states. What usually happens is clicking a button after the first disconnect, for example, will send the command to HA, but the resultant entity state does not update. I see using both a packet capture and developer tools that command messages are being sent from the client and the server is replying with success (see below). Externally-induced state updates are not being sent by the server at all. My hunch would be that this is related to https://github.com/home-assistant/home-assistant-js-websocket/pull/226 but I have not proved this out. I am seeing this both in Chrome 96 and the Android companion app. This is happening with no custom resources installed. The console is free of errors.

### Describe the behavior you expected

State changes and commands always just work.

### Steps to reproduce the issue

1. Open lovelace
2. Wait for the websocket to time out and reconnect
3. Try to use it
...

### What version of Home Assistant Core has the issue?

2021.12.6

### What was the last working version of Home Assistant Core?

2021.12.5

### In which browser are you experiencing the issue with?

Chrome 96

### Which operating system are you using to run this browser?

Linux

### State of relevant entities

```txt
Example exchange of clicking a light entity button after this problem occurs:
From client: {"type":"call_service","domain":"light","service":"turn_off","service_data":{"entity_id":"light.desk_lamp"},"id":19}

From server: {"id": 19, "type": "result", "success": true, "result": {"context": {"id": "450903df694ed37925d8d9629b0c2cd5", "parent_id": null, "user_id": "1bd642b0ed9f410280f622d1d358102b"}}}

No subsequent state_changed event is seen, though the lamp does turn off. Subsequent pushes of the button have no effect since the lamp is already off, but the server does respond success to the light.turn_off call.
```

**Comment by progman32** [comment]
Reproduced here in:

* Firefox (Linux, Android)
* Chromium (Linux)
* Chrome (Android)
* Android Companion app

I've been able to trigger this state by restarting HA. The page doesn't consistently reconnect. I will try 2021.12.4 later and report back (I foolishly didn't take a backup with 2021.12.5).

**Comment by quaqo** [comment]
I think this might be related. 

https://github.com/home-assistant/frontend/issues/11039#issuecomment-1002309735

**Comment by Wolverine80** [comment]
If it's the same problem (switch to turn light on reverts to off, light is on. and vice versa), I get it working again when I Shift-F5 reload the website in Microsoft Edge. No restart of HA necessary But after a few minutes idling, the states won't update again.

**Comment by progman32** [comment]
Downgrade to 2021.12.5 appears to have fixed the issue (`core upgrade --version 2021.12.5` from the HA CLI).

Aside: Gotta say, I am impressed by how easy it is to upgrade and downgrade HA and its components. Kudos to the folks who contributed to this functionality!

**Comment by amigan** [comment]
I managed to mitigate this somewhat by disabling "Automatically close connection
 in the user profile page, at least on my PC. Obviously, it has the luxury to use this setting more than a mobile device. However, I also noticed that it does not attempt to reconnect after the server is restarted; a refresh is required. Just noting this as it seems like it could be germane to fixing the bug.

**Comment by amigan** [comment]
I managed to mitigate this somewhat by disabling "Automatically close connection
 in the user profile page, at least on my PC. Obviously, it has the luxury to use this setting more than a mobile device. However, I also noticed that it any reconnection after the server is restarted does not resubscribe to state changes either; a refresh is required. Just noting this as it seems like it could be germane to fixing the bug.
